### PR TITLE
Create Node and web Heft rig projects.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

File: libraries/debug-certificate-manager/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,5 @@
 {
-  "extends": "./node_modules/@microsoft/rush-stack-compiler-3.5/includes/tsconfig-node.json",
-
+  "extends": "./node_modules/@rushstack/heft-node-rig/profiles/library/tsconfig-base.json",
```
Comment (octogonz) [inline libraries/debug-certificate-manager/tsconfig.json:2]: In the future, based on `rig.json` some tool could now validate this path.

File: libraries/package-deps-hash/package.json [hunk 2]
```diff
@@ -13,12 +13,12 @@
...
   },
   "devDependencies": {
-    "@microsoft/rush-stack-compiler-3.5": "workspace:*",
     "@rushstack/node-core-library": "workspace:*",
     "@rushstack/eslint-config": "workspace:*",
     "@rushstack/heft": "workspace:*",
     "@types/heft-jest": "1.0.1",
-    "@types/node": "10.17.13"
+    "@types/node": "10.17.13",
+    "@rushstack/heft-node-rig": "workspace:*"
```
Comment (octogonz) [inline libraries/package-deps-hash/package.json:21]: Generally I try to keep the dependencies in alphabetical order. Eventually we should use a tool to normalize these files.

File: rigs/heft-node-rig/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+## @rushstack/heft-node-rig
+
+This is a Rig package for [Heft](https://www.npmjs.com/package/@rushstack/heft)
+used for building NodeJS projects.
```
Comment (octogonz) [inline rigs/heft-node-rig/README.md:4]: We should write it as Node.js

File: rigs/heft-node-rig/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+This rig contains a single profile: `library`
+
+To make use of this rig in a project built by Heft, add it as a dependency and include a
+`common/rig.json` file in your project with the following contents:
```
Comment (octogonz) [inline rigs/heft-node-rig/README.md:9]: `config/rig.json`

File: rigs/heft-node-rig/README.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+## @rushstack/heft-node-rig
+
+This is a Rig package for [Heft](https://www.npmjs.com/package/@rushstack/heft)
+used for building NodeJS projects.
+
```
Comment (octogonz) [inline rigs/heft-node-rig/README.md:5]: Add a note saying something like:

> See the [@rushstack/rig-package](#) documentation for more information about rig packages.

File: rigs/heft-node-rig/package.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "@rushstack/heft-node-rig",
+  "version": "0.0.0",
+  "description": "A Heft rig for building Node projects.",
```
Comment (octogonz) [inline rigs/heft-node-rig/package.json:4]: This description misleadingly implies that rigs are  Heft concept.

`A rig package for Node.js projects that build using Heft`

File: rigs/heft-node-rig/README.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+This rig contains a single profile: `library`
```
Comment (octogonz) [inline rigs/heft-node-rig/README.md:6]: What do we use for CLI tools like Rush?

My company also builds Node.js web services (although today they use pretty much the same config as a tool)

Comment (iclanton) [reply rigs/heft-node-rig/README.md:6]: CLI tools aren't really different from any other Node library.

Web services will probably need their own profile.

File: apps/api-documenter/config/rig.json [hunk 8]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (octogonz) [inline apps/api-documenter/config/rig.json:1]: Let's add a comment header to the rig files:
```js
{
  // The "rig.json" file directs tools to look for their config files in an external package.
  // Documentation for this system: https://www.npmjs.com/package/@rushstack/rig-package
  "$schema": "https://developer.microsoft.com/json-schemas/rig-package/rig.schema.json",

  "rigPackageName": "@rushstack/heft-node-rig",
  "rigProfile": "library"
}
```

File: rigs/heft-web-rig/LICENSE [hunk 9]
```diff
@@ -0,0 +1,24 @@
+@rushstack/heft-node-rig
```
Comment (octogonz) [inline rigs/heft-web-rig/LICENSE:1]: Wrong name

File: apps/api-documenter/tsconfig.json [hunk 10]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "./node_modules/@microsoft/rush-stack-compiler-3.5/includes/tsconfig-node.json",
+  "extends": "./node_modules/@rushstack/heft-node-rig/profiles/library/tsconfig-base.json",
```
Comment (octogonz) [inline apps/api-documenter/tsconfig.json:2]: Why is **api-documenter** using `/profiles/library/`? 

It is a CLI tool.

But it does also export a plugin API. Hmm

Comment (octogonz) [reply apps/api-documenter/tsconfig.json:2]: I see -- it is a cyclic dependency

File: rigs/heft-web-rig/LICENSE (hunk 9 above)
Comment (iclanton) [reply rigs/heft-web-rig/LICENSE:1]: Ugh I always miss that.

File: rigs/heft-web-rig/package.json [hunk 11]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "@rushstack/heft-web-rig",
+  "version": "0.0.0",
+  "description": "A Heft rig for building Web projects.",
```
Comment (octogonz) [inline rigs/heft-web-rig/package.json:4]:
```suggestion
  "description": "A rig package for web browser projects that build using Heft",
```

File: rigs/heft-node-rig/README.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
+## @rushstack/heft-node-rig
+
+This is a Rig package for [Heft](https://www.npmjs.com/package/@rushstack/heft)
+used for building Node.js projects.
+
+This rig contains a single profile: `library`
```
Comment (octogonz) [inline rigs/heft-node-rig/README.md:6]:
```suggestion
This rig contains a single profile: `default`
```

**Review by octogonz — APPROVED** [review]

**Pull request merged**